### PR TITLE
Added unit-testing for legacy

### PR DESCRIPTION
**Pull request by afshinmessiah**
I've added unit-testing for sop/legacy module.

### CODE REVIEW

File: tests/test_legacy.py [hunk 1]
```diff
@@ -1,22 +1,249 @@
...
+                ref_dataset_seq[0].file_meta.TransferSyntaxUID = '1.2.3.4.5.6.7.8.9'
+                with self.assertRaises(ValueError):
+                    multiframe_item = LegacyConvertorClass(legacy_datasets=ref_dataset_seq,
+                                                           series_instance_uid=self._output_series_instance_uid,
+                                                           series_number=self._output_instance_number,
+                                                           sop_instance_uid=self._output_sop_instance_uid,
+                                                           instance_number=self._output_instance_number)
+                ref_dataset_seq[0].file_meta.TransferSyntaxUID = tmp_transfer_syntax_uid
+
+    def GenerateCommonDicomDatasetSeries(self, slice_count: int, system: Modality) -> list:
```
Comment (hackermd) [inline tests/test_legacy.py]: Use lower case and underscores rather than CamelCase for method names

File: tests/test_legacy.py [hunk 2]
```diff
@@ -1,22 +1,249 @@
...
+               ('PT', '1.2.840.10008.5.1.4.1.1.128')]
 
-class LegacyConvertedEnhancedCTImage(unittest.TestCase):
+
+class TestLegacyConvertedEnhancedImage(unittest.TestCase):
 
     def setUp(self):
         super().setUp()
+        self._modalities = ('CT', 'MR', 'PET')
+        self._ref_dataset_seq_CT = self.GenerateCommonDicomDatasetSeries(3, Modality.CT)
```
Comment (hackermd) [inline tests/test_legacy.py]: Limit lines to max 80 characters

File: tests/test_legacy.py [hunk 3]
```diff
@@ -34,128 +37,156 @@ def setUp(self):
...
+                        sop_instance_uid=self._output_sop_instance_uid,
+                        instance_number=self._output_instance_number)
                 ref_dataset_seq[0].SOPClassUID = tmp_orig_sop_class_id
 
     def test_mixed_studies(self):
         for m in self._modalities:
             with self.subTest(m=m):
-                LegacyConvertorClass = getattr(sop, "LegacyConvertedEnhanced" + m + "Image")
+                LegacyConvertorClass = \
+                    getattr(sop, "LegacyConvertedEnhanced" + m + "Image")
```
Comment (hackermd) [inline tests/test_legacy.py]:
```suggestion
                LegacyConvertorClass = getattr(
                    sop,
                    "LegacyConvertedEnhanced{}Image".format(m)
                 )
```

Comment (hackermd) [reply tests/test_legacy.py]: When doing multi-line statements put closing bracket on new line and use a separate line for each argument.

Comment (hackermd) [reply tests/test_legacy.py]: Use format strings instead of concatenating substrings using `+`

File: tests/test_legacy.py [hunk 4]
```diff
@@ -1,22 +1,283 @@
...
+from datetime import datetime, timedelta
 import pytest
+import enum
 
 
-class LegacyConvertedEnhancedMRImage(unittest.TestCase):
+class Modality(enum.IntEnum):
+    CT = 0
+    MR = 1
+    PET = 2
```
Comment (fedorov) [inline tests/test_legacy.py]: Would it be more consistent to use PT here in place of PET?

File: tests/test_legacy.py [hunk 5]
```diff
@@ -1,22 +1,283 @@
...
+            self.generate_common_dicom_dataset_series(3, Modality.PET)
+        self._output_series_instance_uid = generate_uid()
+        self._output_sop_instance_uid = generate_uid()
+        self._output_series_number = '1'
+        self._output_instance_number = '1'
 
+    def test_output_attributes(self):
+        for m in self._modalities:
+            with self.subTest(m=m):
+                LegacyConvertorClass = \
```
Comment (fedorov) [inline tests/test_legacy.py]: I _think_ Converter is the proper spelling.

**Comment by fedorov** [comment]
I have not tried to run, but this looks pretty good to me, definitely very good for the first tests! Thank you @afshinmessiah!

File: tests/test_legacy.py (hunk 4 above)
Comment (afshinmessiah) [reply tests/test_legacy.py]: Though this would be more readable. Sure ...

File: tests/test_legacy.py (hunk 5 above)
Comment (afshinmessiah) [reply tests/test_legacy.py]: Both [spellings ](https://www.merriam-webster.com/dictionary/converter)look correct. The converter is more popular though. Will change it

File: tests/test_legacy.py (hunk 3 above)
Comment (afshinmessiah) [reply tests/test_legacy.py]: Sure. Will do.

File: tests/test_legacy.py [hunk 6]
```diff
@@ -1,22 +1,297 @@
...
+# class LegacyConvertedEnhancedCTImage(unittest.TestCase):
+#
+#     def setUp(self):
+#         super().setUp()
+#
+#
+# class LegacyConvertedEnhancedPETImage(unittest.TestCase):
+#
+#     def setUp(self):
+#         super().setUp()
```
Comment (hackermd) [inline tests/test_legacy.py]:
```suggestion
```

**Pull request merged**